contracts/DNSSEC.sol

```diff
@@ -164,6 +162,37 @@ contract DNSSEC is Owned {
         emit RRSetUpdated(name);
     }
 
+    /**
+     * @dev Deletes a RR from the oracle.
+     *
+     * @param dnsclass The DNS class (1 = CLASS_INET) of the records being inserted.
+     * @param nsecname which contains the next authorative record
+     * @param deletetype The DNS record type to delete.
+     * @param deletename which you want to delete
+     * 
+     */
+    function deleteRRSet(uint16 dnsclass, bytes nsecname, uint16 deletetype, bytes deletename) public {
+        RRSet storage result = rrsets[keccak256(nsecname)][DNSTYPE_NSEC][dnsclass];
+        if(int(result.inserted) == 0) return;
+        for(RRUtils.RRIterator memory iter = result.rrs.iterateRRs(0); !iter.done(); iter.next()) {
+            if (iter.dnstype == DNSTYPE_NSEC){
+                bytes memory rdata = iter.rdata();
+                uint nextNameLength = rdata.nameLength(0);
+                uint rDataLength = rdata.length;
+                bytes memory nextName = rdata.substring(0,nextNameLength);
+                int compareResult = deletename.compareLabel(nextName);
+                if (compareResult < 0){
+                    delete rrsets[keccak256(deletename)][deletetype][dnsclass];
+                }else if (compareResult == 0) {
```

Comment:
This code should run if `nsecname == deleteName`, not if `nextName == deleteName`.